### PR TITLE
tests: fix error in TestGlobalAndLocalTSO

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: Song Gao <disxiaofei@163.com>

### What problem does this PR solve?

ref https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/pd_test/detail/pd_test/19200/pipeline/

TestGlobalAndLocalTSO is not stable. This request fix it.

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Code changes

- Has configuration change
- Has HTTP API interfaces change (Don't forget to [add the declarative for API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tidb-ansible`](https://github.com/pingcap/tidb-ansible):
- Need to cherry-pick to the release branch

### Release note

* No release note

### CODE REVIEW

**Review by JmPotato — APPROVED** [review]
LGTM

File: client/base_client.go [hunk 1]
```diff
@@ -145,7 +146,13 @@ func (c *baseClient) leaderLoop() {
 		case <-ctx.Done():
 			return
 		}
-
+		skipUpdateLeader := false
+		failpoint.Inject("skipUpdateLeader", func() {
+			skipUpdateLeader = true
```
Comment (rleungx) [inline client/base_client.go:151]: How about directly continue?

Comment (Yisaer) [reply client/base_client.go:151]: we can't directly continue in this fuction.

**Review by rleungx — APPROVED** [review]

**Pull request merged**